### PR TITLE
Support TLS; Reduce the need of config.toml in integration tests

### LINKED ISSUE #262: Support TLS for security
## Feature Request

**Is your feature request related to a problem? Please describe:**

When PD/TiKV/TiDB configures with TLS for connection, TiDB Lightning may not work and also no security.

**Describe the feature you'd like:**

Support TLS for connecting with PD/TiKV/TiDB and TiKV Importer.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Optimization:**

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

Fix #262.

### What is changed and how it works?

1. Code changes:

    * Added the `[security]` section in the config to read CA, cert and key. These are used to construct the standard [`*tls.Config`](https://pkg.go.dev/crypto/tls?tab=doc#Config).
    * This config is used to:
        * [Add a TLS layer](https://pkg.go.dev/crypto/tls?tab=doc#NewListener) on top of the web interface / metrics HTTP server
        * Encrypt gRPC communication with Importer and TiKV (via [WithTransportCredentials](https://pkg.go.dev/google.golang.org/grpc?tab=doc#WithTransportCredentials))
        * Encrypt HTTP communication with PD and TiDB status port
        * Encrypt MySQL communication with TiDB (via the [`tls` parameter in the DSN](https://github.com/go-sql-driver/mysql#tls))
    * All of these operations are grouped into a struct `common.TLS` for simplified management. This struct mainly acts as an `http.Client` to fetch JSON objects, plus methods to produce options for securing gRPC and MySQL protocols.

2. Test changes:

    * The entire integration test is changed to use TLS for all communications.
    * The certs need be to pass into every `tidb-lightning` and `tidb-lightning-ctl` invocations. To simplify future changes, the `run_lightning` and `run_lightning_ctl` helper scripts now define most common settings in the command line.
    * Existing `config.toml`s are simplified to retain only the essential settings.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
    * check that web interface does serve HTTPS if required.

Side effects

Related changes

 - Need to update the documentation
    * Describe the new config and CLI parameters
 - Need to update the `tidb-ansible` repository
    * Copy the global TLS settings to config
 - Need to be included in the release note
    * Note TLS support

### CODE REVIEW

**Comment by kennytm** [comment]
PTAL @july2993 @3pointer

File: lightning/lightning.go [hunk 1]
```diff
@@ -66,9 +67,15 @@ func New(globalCfg *config.GlobalConfig) *Lightning {
 		os.Exit(1)
 	}
 
+	tls, err := common.NewTLS(globalCfg.Security.CAPath, globalCfg.Security.CertPath, globalCfg.Security.KeyPath, globalCfg.App.StatusAddr)
+	if err != nil {
+		log.L().Fatal("failed to load TLS certificates", zap.Error(err))
```
Comment (3pointer) [inline lightning/lightning.go]: if `NewTLS` failed, should exit immediately？

Comment (kennytm) [reply lightning/lightning.go]: yes, for global config, currently it exits immediately

File: lightning/common/security.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+	// Append the certificates from the CA
+	if !certPool.AppendCertsFromPEM(ca) {
+		return nil, errors.New("failed to append ca certs")
+	}
+
+	inner := &tls.Config{
+		Certificates: certificates,
+		RootCAs:      certPool,
+		NextProtos:   []string{"h2", "http/1.1"},
```
Comment (july2993) [inline lightning/common/security.go]: what's the usage of this

Comment (kennytm) [reply lightning/common/security.go]: to allow HTTP/2.

**Comment by kennytm** [comment]
PTAL @july2993 @3pointer again thanks.

File: lightning/common/security.go (hunk 2 above)
Comment (july2993) [reply lightning/common/security.go]: https://pkg.go.dev/net/http?tab=doc#Serve

 HTTP/2 support is only enabled if the Listener returns *tls.Conn connections and they were configured with "h2" in the TLS Config.NextProtos.

**Comment by 3pointer** [comment]
there are two kind of security, `global security` and `tidb security`, if pass different security here, it seems that `importer` and `tikv` will use `global security` and `tidb`, `pd` will use another. shoud `tidb`,`pd`,`tikv` use the same one?

**Comment by kennytm** [comment]
Only SQL connection uses `[tidb.security]` (corresponding to `ssl-ca/cert/key` in TiDB). Everything else should use `[security]` (corresponding to `cluster-ssl-ca/cert/key` in TiDB). If not it's a bug.

**Pull request merged**